pkg/mounter/mount_windows.go

```diff
@@ -128,30 +118,42 @@ func (m NodeMounter) FormatAndMountSensitiveWithFormatOptions(source string, tar
 // Command to determine ref count would be something like:
 // Get-Volume -UniqueId "\\?\Volume{7c3da0c1-0000-0000-0000-010000000000}\" | Get-Partition | Select AccessPaths
 func (m NodeMounter) GetDeviceNameFromMount(mountPath string) (string, int, error) {
-	proxyMounter, ok := m.SafeFormatAndMount.Interface.(*CSIProxyMounter)
-	if !ok {
-		return "", 0, fmt.Errorf("failed to cast mounter to csi proxy mounter")
-	}
-	deviceName, err := proxyMounter.GetDeviceNameFromMount(mountPath, "")
-	if err != nil {
-		// HACK change csi-proxy behavior instead of relying on fragile internal
-		// implementation details!
-		// if err contains '"(Get-Item...).Target, output: , error: <nil>' then the
-		// internal Get-Item cmdlet didn't fail but no item/device was found at the
-		// path so we should return empty string and nil error just like the Linux
-		// implementation would.
-		pattern := `(Get-Item -Path \S+).Target, output: , error: <nil>|because it does not exist`
-		matched, matchErr := regexp.MatchString(pattern, err.Error())
-		if matched {
-			return "", 0, nil
+	switch proxyMounter := m.SafeFormatAndMount.Interface.(type) {
+	case *CSIProxyMounterV2:
+		deviceName, err := proxyMounter.GetDeviceNameFromMount(mountPath, "")
+		if err != nil {
+			return handleGetDeviceNameFromMountError(err)
 		}
-		err = fmt.Errorf("error getting device name from mount: %v", err)
-		if matchErr != nil {
-			err = fmt.Errorf("%v, and error matching pattern %q: %v", err, pattern, matchErr)
+		return deviceName, 1, nil
+	case *CSIProxyMounter:
+		deviceName, err := proxyMounter.GetDeviceNameFromMount(mountPath, "")
+		if err != nil {
+			// HACK change csi-proxy behavior instead of relying on fragile internal
+			// implementation details!
+			// if err contains '"(Get-Item...).Target, output: , error: <nil>' then the
+			// internal Get-Item cmdlet didn't fail but no item/device was found at the
+			// path so we should return empty string and nil error just like the Linux
+			// implementation would.
```

Comment:
Can we move this comment to before the case? Because it applies to both the V1 and V2 paths as I understand it

Reply:
Moved comment to the top above the case statement ✅